src/storage/hummock_sdk/src/table_watermark.rs

```diff
@@ -102,79 +102,54 @@ impl TableWatermarksIndex {
         self.read_watermark(vnode, HummockEpoch::MAX)
     }
 
-    pub fn range_watermarks(
+    pub fn rewrite_range_with_table_watermark(
         &self,
         epoch: HummockEpoch,
         key_range: &mut TableKeyRange,
```

Comment:
just question: why not return a new key_range after rewrite instead of "mut ref "?

Reply:
Because we also do in-place update prior to this PR. I am okay both ways so I can also change that if you think it is important.